From 2faa025f1075fbac3fbd997b86dc17ceaad148da Mon Sep 17 00:00:00 2001
From: Eduardo Ferreira <eduardo.barbosa@toradex.com>
Date: Mon, 29 Jan 2024 09:40:51 -0300
Subject: [PATCH] zipfile.py: do not raise error if date_time < 1980

The build of some Python packages (e.g.: setuptools) is failing with the
following error:

ValueError: ZIP does not support timestamps before 1980

Prevent that by forcing the date_time when (for some reason) the value is
less than 1980.

Upstream-Status: Inappropriate [Torizon specific]

Signed-off-by: Sergio Prado <sergio.prado@toradex.com>
Signed-off-by: Eduardo Ferreira <eduardo.barbosa@toradex.com>
---
 Lib/zipfile/__init__.py | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/Lib/zipfile/__init__.py b/Lib/zipfile/__init__.py
index 420d2f8..17a38ad 100644
--- a/Lib/zipfile/__init__.py
+++ b/Lib/zipfile/__init__.py
@@ -386,6 +386,9 @@ class ZipInfo (object):
     def __init__(self, filename="NoName", date_time=(1980,1,1,0,0,0)):
         self.orig_filename = filename   # Original file name in archive
 
+        if date_time[0] < 1980:
+            date_time=(1980,1,1,0,0,0)
+
         # Terminate the file name at the first null byte and
         # ensure paths always use forward slashes as the directory separator.
         filename = _sanitize_filename(filename)
-- 
2.34.1

